### PR TITLE
Feature/upgrade to marionette v3

### LINKED ISSUE #139: Toolkit App should extend from MnV3.App
Also we'll need to stop setting `_isDestroyed` as `MnV3.Object` will handle that

### PR DESCRIPTION
**Pull request by TerrenceLJones**
This resolves #139

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -74,8 +74,8 @@
     "exportVarName": "Marionette.Toolkit"
   },
   "peerDependencies": {
-    "backbone.marionette": "^2.4.5",
-    "backbone": "^1.3.2",
+    "backbone.marionette": "3.0.0-pre.5",
```
Comment (paulfalgout) [inline package.json]: This can just be set to "^3.0.0"
I think we can be safe with setting the peer Deps on backbone to `^1.3.0` as well.

**Comment by paulfalgout** [comment]
This PR needs to adapt to the new features of Mn v3 as well.
`Application` now manages `_isDestroyed` and has an `isDestroyed`
Additionally `Application` has a `region` API that should most likely be utilized by the component.

**Comment by paulfalgout** [comment]
Other thing.. now this lib will depend on Marionette v3 which is probably documented in the readme?  I suspect there are other things it extends now as well?

**Comment by TerrenceLJones** [comment]
Can you explain what you mean by *"suspect there are other things it extends now as well"*?

**Comment by paulfalgout** [comment]
Sorry commenting while childrening.  Docs probably say "extends Marionette.Object" somewhere..  Or something like it

**Comment by TerrenceLJones** [comment]
Alt Text

**Pull request merged**